src/export.h

```diff
@@ -30,8 +30,40 @@
 #endif							/* __GNUC__ */
 #endif
 
+#define TS_PREFIX(fn) \
+	TS_CAT(ts_, fn)
+
+/*
+ * Functions exposed to postgres always have the signature
+ *   `PGDLLEXPORT Datum function(PG_FUNCTION_ARGS)`
+ * and function name should always be of the form `ts_realname`
+ * We define the following macros to assist with this
+ */
+
+/*
+ * Used as to declare and define publically-accessible functions, e.g.
+ *
+ *    extern TS_FUNCTION(realname);
+ *
+ * or
+ *
+ *    TS_FUNCTION(realname)
+ *    {
+ *    	// body
+ *    }
+ *
+ */
+#define TS_FUNCTION(fn) \
```

Comment:
If we're doing this, why can't we do the export in the same macro so that you don't need to do both `TS_FUNCTION()` and `TS_FUNCTION_INFO_V1()`?

Reply:
Just to be clear, the idea would be we just have?:
```C
TS_FUNCTION_HEADER(realname) // macro name is just a placeholder
{
    // function body
}
```
I like it! And it'd stop bugs when people put FUNCTION_INFO in a header.
Even with this, we'd still need something like the `TS_FUNCTION` from this PR for headers, since creating multiple versions of the FUNCTION_INFO causes linker errors, but it should be more rare.